guides/source/layout.html.erb

```diff
@@ -39,59 +45,83 @@
         <li class="more-info"><a href="https://github.com/rails/rails">Contribute on GitHub</a></li>
       </ul>
     </div>
-  </div>
-  <div id="header">
+  </nav>
+  <header id="page_header">
     <div class="wrapper clearfix">
-      <div class="header-logo"><a href="index.html" title="Return to home page">Guides.rubyonrails.org</a></div>
-      <ul class="nav">
-        <li><a class="nav-item" href="index.html">Home</a></li>
-        <li class="guides-index guides-index-large">
-          <a href="index.html" id="guidesMenu" class="guides-index-item nav-item">Guides Index</a>
-          <div id="guides" class="clearfix" style="display: none;">
-            <hr />
-            <dl class="guides-section-container">
-              <% documents_by_section.each do |section| %>
-                <div class="guides-section">
-                  <dt><%= section['name'] %></dt>
+      <nav id="feature_nav"> <%# we use nav as an element to also provide the aria role and improve semantic meaning. %>
+        <div class="header-logo">
+          <a href="index.html" title="Return to home page">Guides</a>
+          <span id="version_switcher">
+            Version:
+            <select class="guides-version">
+              <option value="https://edgeguides.rubyonrails.org/" selected>Edge</option>
```

Comment:
Why is Edge the selected? We don't want people to use the documentation of a version of Rails that wasn't released yet. Of couse it should be the selected based on which version we are on, so this should be dynamic somehow.

Reply:
So on build, it builds edge. 7.1 doesn't exist on `rake guides:generate`. Is there a way to build 7.1 locally to test this?

@carlosantoniodasilva did your change to guides.js set this to update to whatever is active? I'm fine manually setting selected to 7.1 for now since loading the domain would load that, but ideally it works as @rafaelfranca describes and just selects which version it is on.

Reply:
`@edge` / `@version` (from the currently removed version badge above) can probably help with this.

Reply:
I'm sure there's a better way to do this, but my first "sledgehammer" way to get this in is a conditional on each `option`, e.g.:

```
<option value="https://edgeguides.rubyonrails.org/" <% if @edge == true %>selected<% end %> >Edge</option>
```

which seems dirty. Is there a better way? Maybe an interator over an array of versions that build each option?

Reply:
Yeah, edge probably needs individual handling anyway, but for the rest, collapsing the full links into an array of version numbers seems both helpful and independently desirable :+1:

Reply:
We don't need to refactor the list into Ruby generated list for now, as there's a similar list on the [intro for non-edge guides here](https://github.com/rails/rails/blob/94faed4dd4c915829afc2698e055a0265d9b5773/guides/source/_welcome.html.erb#L19-L32). We can circle back on refactoring those two lists later.

As for `selected`, we could do it in Ruby when the page is rendered via `@edge / @version`, but I thought we could do it in JS similar to how the current "guide" gets selected on the mobile version that shows a select:

https://github.com/rails/rails/blob/94faed4dd4c915829afc2698e055a0265d9b5773/guides/assets/javascripts/guides.js#L69

That's what I tried to accomplish in this bit of JS: https://github.com/rails/rails/pull/51341/files/bef423f59ec06f346ec64ac35a143f1dd7a9ecd9#diff-edcbfbf5f4b023773d5793ae6ac38eac5a5ec6213c25f0ca6b7d9a8e008621c7R67-R70

I think we could just remove the default `selected` from Edge here, if that JS works it should always show the current version based on the hostname/path.

Reply:
So I think first is remove `selected` and let JS handle that, and second is add the version numbers to an array and build the option line programmatically (just static HTML, but easier editing in the future for adding/removing versions). Does that sound like a plan?

Reply:
https://github.com/rails/rails/pull/51341/files/bef423f59ec06f346ec64ac35a143f1dd7a9ecd9#diff-edcbfbf5f4b023773d5793ae6ac38eac5a5ec6213c25f0ca6b7d9a8e008621c7R72-R74 probably breaks arrow-key changing of the selector as it's unlikely the `visit` call will retain the focus. A list of links would probably be more appropriate.

That said, personally I don't like to defer to JavaScript to select the right option, if the right option is known when the HTML is generated.

Reply:
@SleeplessByte it shouldn't break arrow navigation, since the "change" event only triggers when you actually select one (i.e. press enter on an option)

@rafaelfranca we do. I implemented that following the other select though, for the current guide, even thinking I could've reused the same "change" event... but I can change.

Reply:
@carlosantoniodasilva no that is a misconception. Tab to the `select` element and do not open it. Now press down arrow. This triggers a change event.

Reproducible for chromium based browsers and Firefox on Windows.

Reply:
@SleeplessByte it might be and I just don't know about it 👍 . My quick test on Chrome / Safari on a Mac, even using tab + down, displays the select options:

<details><summary>Details</summary>
<p>

https://github.com/rails/rails/assets/26328/310acb38-a207-4c9f-8f97-d290d121e0da

</p>
</details>

Reply:
Here's a suggestion on how to set the selected version upfront during guides generation, as well as using a list of versions to generate the options: https://github.com/jathayde/rails/pull/2

Reply:
@carlosantoniodasilva Ah, that explains where you are coming from. 
Here is a video to show the behaviour I described:

**Chrome**:

https://github.com/rails/rails/assets/1964376/383a348d-537d-4b5c-a092-bc2f8f2beafc

**Firefox**: 

https://github.com/rails/rails/assets/1964376/cd27bed9-49b6-4881-a54d-46e8ef388aaf



The keys I pressed are:
- Tab
- Tab
- Down arrow

Reply:
@SleeplessByte thanks, yeah not sure why the behavior is so different. The same problem in theory exists on the guide selector on smaller screens, likely something we'd have to look into improving with other accessibility things.

Reply:
(either way, I think we oughtta ignore it for now and group with https://github.com/rails/rails/pull/51341#issuecomment-2007717282 like Amanda suggests!)